vita3k/renderer/src/gl/renderer.cpp

```diff
@@ -486,9 +488,15 @@ void get_surface_data(GLContext &context, size_t width, size_t height, size_t st
         glReadPixels(0, 0, static_cast<GLsizei>(width), static_cast<GLsizei>(height), GL_RED, GL_UNSIGNED_BYTE, pixels);
         break;
     case SCE_GXM_COLOR_FORMAT_U2F10F10F10_ABGR:
+    case SCE_GXM_COLOR_FORMAT_F10F10F10U2_RGBA:
     case SCE_GXM_COLOR_FORMAT_U2U10U10U10_ABGR:
+    case SCE_GXM_COLOR_FORMAT_U10U10U10U2_RGBA:
         glReadPixels(0, 0, static_cast<GLsizei>(width), static_cast<GLsizei>(height), GL_RGBA, GL_UNSIGNED_INT_2_10_10_10_REV, pixels);
         break;
+    case SCE_GXM_COLOR_FORMAT_F10F11F11_BGR:
+    case SCE_GXM_COLOR_FORMAT_F11F11F10_RGB:
+        glReadPixels(0, 0, static_cast<GLsizei>(width), static_cast<GLsizei>(height), GL_BGR, GL_UNSIGNED_INT_10F_11F_11F_REV, pixels);
```

Comment:
One is BGR the other is RGB but you're using the same readback format. Now way for both to be right.

Reply:
@Zangetsu38 

example
SCE_GXM_COLOR_FORMAT_F11F11F10_RGB -> GL_UNSIGNED_INT_10F_11F_11F_REV
SCE_GXM_COLOR_FORMAT_F10F11F11_BGR -> GL_UNSIGNED_INT_10F_11F_11F

PS: I don't know which one is REV

Reply:
Only this value existe, so no have choice

Reply:
ReadPixels spec only defines the REV version, but this means post-transformation is required on the CPU client. The data is wrong either way.

Reply:
OK so how set it correctly ? I have no idea here 😢